### PR TITLE
IC-989: Use Wiremock to selectively mock interventions service in local development

### PR DESCRIPTION
**Pull request by lawrence-forooghian**
## What does this pull request do?

Puts a Wiremock instance in front of the interventions service in local development, and provides a way for us to configure mocked responses.

## What is the intent behind these changes?

To allow us to use the real interventions service whilst providing mocked responses for functionality that has not yet been built.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -35,6 +35,13 @@ Navigate to `http://localhost:3000` and log in:
 - To log in as a service provider user, use HMPPS Auth dev credentials e.g. `AUTH_ADM/password123456`
 - To log in as a probation practitioner user, use [Community API dev credentials](https://github.com/ministryofjustice/community-api/blob/main/src/main/resources/schema.ldif) e.g. `bernard.beaks/secret`.
 
+### Mocking out the interventions service
+
+In development mode, you might want to selectively mock out some calls to the
+interventions service. For example, for endpoints that have not yet been built.
+
+You can do this by configuring mocks in [`mocks.ts`](mocks.ts).
```
Comment (Gweaton) [inline README.md:43]: Yeah, I think this is a good idea for now, as it'd be a pain to set up locally on everyone's machine. As long as we remember to remove them!

**Review by Gweaton — COMMENTED** [review]
I've had a quick first pass at this, as I've run out of time. Have left a couple of comments but looks good so far, and a lot of work, so well done!

File: wiremock_mappings/interventions_service_proxy.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply wiremock_mappings/interventions_service_proxy.json:1]: The difference is that one of them shouldn't exist 😬 - will remove, thanks

File: README.md (hunk 1 above)
Comment (lawrence-forooghian) [reply README.md:43]: 👍

File: integration_tests/mockApis/auth.ts [hunk 3]
```diff
@@ -0,0 +1,198 @@
...
+      auth_source: authSource,
+      authorities: [],
+      jti: '83b50a10-cca6-41db-985f-e87efb303ddb',
+      client_id: 'interventions',
+    }
+
+    return jwt.sign(payload, 'secret', { expiresIn: '1h' })
+  }
+
+  async getLoginUrl(): Promise<string> {
```
[the remark this answers is not in the document]
Comment (lawrence-forooghian) [reply integration_tests/mockApis/auth.ts]: It does work! I've changed it. I still don't 100% understand why it works but that can wait until another time (`this` remains a bit of a mystery to me in general in JS).

**Review by Gweaton — APPROVED** [review]
Nice one.

**Pull request merged**